doc/pac/Port Access Control.md

```diff
@@ -0,0 +1,844 @@
+
+### 3.4.4 hostapdmgrd
+hostapdmgr reads hostapd specific configuration from SONiC DBs and populates the hostapd.conf. It further notifies the hostapd to re-read the configuration file.
+
+### 3.4.5 Interaction between modules
+
+*hostapd(802.1X)*   
+
+hostapd comes to know of an 802.1x client attempting authentication via an EAP exchange. It informs pacd of this client by conveying the client MAC. If the authentication method selected by pacd is 802.1X, pacd sends an event to hostapd for authenticating the user. The client is however authenticated via MAB If the authentication method selected here is MAB.
+
+hostapd informs pacd about the result of the authentication. hostapd also passes all the authorization parameters it receives from the RADIUS Server to the pacd. These are used for configuring the switch to allow authenticated client traffic.
+
+*mabd(MAB)*   
+
+When user or client tries to authenticate and the method selected is MAB, the pacd sends an event to mabd for authenticating the user. The client’s MAC address is sent to mabd for the same. 
+
+pacd learns client’s MAC address through an hardware rule to copy-to-CPU the packets from unknown source MAC addresses.
+
+mabd informs pacd about the result of the authentication. mabd also passes all the authorization parameters it receives from the RADIUS Server to the pacd. These are used for configuring the NAS to allow authenticated client traffic.
+
+
+## 3.5 SyncD
+
+No specific changes are needed in syncd for PAC.
+
+## 3.6 SAI
+
+Existing SAI attributes are used for this implementation and there is no new SAI attribute requirement.
+
+### 3.6.1 Host interface traps
+
+Leveraged **SAI_HOSTIF_TRAP_TYPE_EAPOL** to trap EAP packets (Ethertype - 0x888E) to the CPU.   
+
+
+### 3.6.2 Bridge port learning modes
+PAC uses the following bridge port learning modes to drop/trap all unknown source MAC packets.   
+- SAI_BRIDGE_PORT_FDB_LEARNING_MODE_DROP
+- SAI_BRIDGE_PORT_FDB_LEARNING_MODE_HW
+- SAI_BRIDGE_PORT_FDB_LEARNING_MODE_CPU_TRAP
+
+
+
+### 3.6.3 FDB
+PAC uses **SAI_FDB_ENTRY_ATTR_PACKET_ACTION** with **SAI_PACKET_ACTION_DROP** to put the static FDB entry in discard state.   
+**SAI_PACKET_ACTION_FORWARD** is used to put the static FDB entry into forwarding state post successful client authentication.   
+
+
+
+
+## 3.7 Manageability
+
+### 3.7.1 Yang Model
+Yang Models are available for managing PAC, hostapd and MAB modules. SONiC YANG is added as part of this contribution.
+
+### 3.7.2 Configuration Commands
+
+The following commands are used to configure PAC.  
+
+| CLI Command                              | Description                              |
+| :--------------------------------------- | :--------------------------------------- |
+| config authentication port-control interface <auto \| force-authorized \| force-unauthorized \>  <interface\> | This command configures the authentication mode to use on the specified interface. Default is force-authorized. |
```

Comment:
Please use Click auto-generation tool to generate the CLIs. Refer below for more information.
https://github.com/sonic-net/SONiC/pull/780

Reply:
OK. Thanks.